include/wlr/types/wlr_tablet_pad.h

```diff
@@ -20,16 +20,40 @@ struct wlr_tablet_pad {
 		struct wl_signal button;
 		struct wl_signal ring;
 		struct wl_signal strip;
+		struct wl_signal attach_tablet; //struct wlr_tablet_tool
 	} events;
 
+	size_t button_count;
+	size_t ring_count;
+	size_t strip_count;
+
+	struct wl_list groups; // wlr_tablet_pad_group::link
+	struct wl_list paths; // wlr_tablet_path::link
```

Comment:
Is it possible to use a regular array for paths? (`char **paths` + `size_t paths_len`)

Reply:
Probably I'd say. It's nice/easy on libinput, semi-nice on wayland backend (have to aggregate).
I'd prefer the old `list_t` (I think `wlr_list` now) over blank array.